app/src/main/java/com/github/se/polyfit/viewmodel/meal/MealViewModel.kt

```diff
@@ -27,14 +27,26 @@ class MealViewModel @Inject constructor(private val mealRepo: MealRepository) :
     get() = _meal
 
   private val _isComplete: StateFlow<Boolean> =
-      _meal.map { it.isComplete() }.stateIn(GlobalScope, SharingStarted.Eagerly, false)
+      _meal.map { it.isComplete() }.stateIn(viewModelScope, SharingStarted.Eagerly, false)
   val isComplete: StateFlow<Boolean>
     get() = _isComplete
 
   fun setMealData(meal: Meal) {
     _meal.value = meal
   }
 
+  fun setMealData(mealId: Long) {
```

Comment:
What are your thoughts of making this part of the `MealViewModel` init flow? If we are no longer reusing viewmodels, we will never need this function other than when the ViewModel is created. So, we can make an optional parameter in the constructor for mealId, and then run this there

Reply:
That's a great idea. I am a bit hesitant to implement that because all the arguments are injected by Hilt. I'll give it a try to see if I can make it work.
Update : 
The issue is that because the ViewModel is created by Hilt, we cannot dynamically pass arguments to it when it's created. So, I don't think this is possible. I did clean up the implementation a bit by changing the argument of the function from Long to Long? and removing the if condition in the `AddMealFlow`.